### PR TITLE
Disable service workers by default

### PR DESCRIPTION
**Pull request by chuckh**
This is a first pass at a pull request  that does the following:
1. Disables services workers by default by using comments and removing precache
1. Updates readme.md to reflect these changes.

This addresses #80.

### CODE REVIEW

**Comment by chuckh** [comment]
@addyosmani still need to cleanup the merge commits, but wanted to start the review process.

**Comment by chuckh** [comment]
@addyosmani after deciding what to do with this pull request, should a new release be created?

**Comment by chuckh** [comment]
@addyosmani or better yet possibly for the next PSK release of 1.0.3.

**Comment by robdodson** [comment]
lgtm

**Comment by addyosmani** [comment]
I think we should land this as part of 1.0.3. Thanks for working on this @chuckh!

File: README.md [hunk 1]
```diff
@@ -6,10 +6,10 @@
 ### Included out of the box:
 
 * [Polymer](http://polymer-project.org), [Paper](https://elements.polymer-project.org/browse?package=paper-elements) and [Iron](https://elements.polymer-project.org/browse?package=iron-elements) elements
-* [Material Design](http://www.google.com/design/spec/material-design/introduction.html) layout 
+* [Material Design](http://www.google.com/design/spec/material-design/introduction.html) layout
 * Routing with [Page.js](https://visionmedia.github.io/page.js/)
 * Unit testing with Web Component Tester
-* Offline-first setup through [Platinum](https://elements.polymer-project.org/browse?package=platinum-elements) Service Worker elements
+* Optional Offline-first setup through [Platinum](https://elements.polymer-project.org/browse?package=platinum-elements) Service Worker elements
```
Comment (arthurvr) [inline README.md]: Does `Offline` still need a capital letter?

File: README.md [hunk 2]
```diff
@@ -118,9 +118,54 @@ Polymer uses [Bower](http://bower.io) for package management. This makes it easy
...
+Polymer Starter Kit offers an optional offline-first experience thanks to Service Worker and the [Platinum Service Worker elements](https://github.com/PolymerElements/platinum-sw). New to Service Worker? Read the following [introduction](http://www.html5rocks.com/en/tutorials/service-worker/introduction/) to understand how it works.
+
+Our optional default offline setup should work well for relatively simple applications. For more complex apps, we recommend learning how Service Worker works so that you can make the most of the Platinum Service Worker element abstractions.
+
+### Enable Service Worker support?
+
+To enable Service Worker support for Polymer Starter Kit project use these 3 steps:
+
+1. Uncomment Service Worker code in index.html
+  ```c
```
Comment (arthurvr) [inline README.md]: `c`?

File: README.md [hunk 3]
```diff
@@ -118,9 +118,54 @@ Polymer uses [Bower](http://bower.io) for package management. This makes it easy
...
+                        skip-waiting
+                        on-service-worker-installed="displayInstalledToast">
+    <platinum-sw-cache default-cache-strategy="networkFirst"
+                       precache-file="precache.json">
+    </platinum-sw-cache>
+  </platinum-sw-register>
+  -->
+  ```
+1. Uncomment Service Worker code in elements.html
+```c
```
Comment (arthurvr) [inline README.md]: `c`?

File: README.md [hunk 4]
```diff
@@ -118,9 +118,54 @@ Polymer uses [Bower](http://bower.io) for package management. This makes it easy
...
+<!-- Uncomment next block to enable Service Worker Support (2/2) -->
+<!--
+<link rel="import" href="../bower_components/platinum-sw/platinum-sw-cache.html">
+<link rel="import" href="../bower_components/platinum-sw/platinum-sw-register.html">
+-->
+```
+1. Add 'precache' to the list in the 'default' gulp task like below. [(gulpfile.js)](https://github.com/PolymerElements/polymer-starter-kit/blob/master/gulpfile.js)
+  ```c
```
Comment (arthurvr) [inline README.md]: `c`?

File: README.md [hunk 5]
```diff
@@ -118,9 +118,54 @@ Polymer uses [Bower](http://bower.io) for package management. This makes it easy
...
+                        clients-claim
+                        skip-waiting
+                        on-service-worker-installed="displayInstalledToast">
+    <platinum-sw-cache default-cache-strategy="networkFirst"
+                       precache-file="precache.json">
+    </platinum-sw-cache>
+  </platinum-sw-register>
+  -->
+  ```
+1. Uncomment Service Worker code in elements.html
```
Comment (arthurvr) [inline README.md]: `2.` right?

File: README.md [hunk 6]
```diff
@@ -118,9 +118,54 @@ Polymer uses [Bower](http://bower.io) for package management. This makes it easy
...
+```c
+<!-- Uncomment next block to enable Service Worker Support (2/2) -->
+<!--
+<link rel="import" href="../bower_components/platinum-sw/platinum-sw-cache.html">
+<link rel="import" href="../bower_components/platinum-sw/platinum-sw-register.html">
+-->
+```
+1. Add 'precache' to the list in the 'default' gulp task like below. [(gulpfile.js)](https://github.com/PolymerElements/polymer-starter-kit/blob/master/gulpfile.js)
```
Comment (arthurvr) [inline README.md]: `3.` right?

File: README.md [hunk 7]
```diff
@@ -118,9 +118,54 @@ Polymer uses [Bower](http://bower.io) for package management. This makes it easy
...
+  ```c
+  // Build Production Files, the Default Task
+  gulp.task('default', ['clean'], function (cb) {
+    runSequence(
+      ['copy', 'styles'],
+      'elements',
+      ['jshint', 'images', 'fonts', 'html'],
+      'vulcanize', 'precache',
```
Comment (arthurvr) [inline README.md]: more consistent would be if these were on 2 separate lines

File: README.md [hunk 8]
```diff
@@ -160,15 +205,14 @@ In order to guarantee that the latest version of your Service Worker script is b
...
-* Remove the two Platinum Service Worker elements (platinum-sw/..) in [app/elements/elements.html](https://github.com/PolymerElements/polymer-starter-kit/blob/master/app/elements/elements.html)
-* Remove references to the platinum-sw elements from your application [index](https://github.com/PolymerElements/polymer-starter-kit/blob/master/app/index.html). 
+If for any reason you need to disable Service Worker support after previously enabling it, you can remove it from your Polymer Starter Kit project using these 4 steps:
 
-You will also want to navigate to `chrome://serviceworker-internals` and unregister any Service Workers registered by Polymer Starter Kit for your app just in case there's a copy of it cached. 
+1. Remove references to the platinum-sw elements from your application [index](https://github.com/PolymerElements/polymer-starter-kit/blob/master/app/index.html).
+1. Remove the two Platinum Service Worker elements (platinum-sw/..) in [app/elements/elements.html](https://github.com/PolymerElements/polymer-starter-kit/blob/master/app/elements/elements.html)
```
Comment (arthurvr) [inline README.md]: `2.`

File: README.md [hunk 9]
```diff
@@ -160,15 +205,14 @@ In order to guarantee that the latest version of your Service Worker script is b
...
+1. Remove 'precache' from the list in the 'default' gulp task ([gulpfile.js](https://github.com/PolymerElements/polymer-starter-kit/blob/master/gulpfile.js))
```
Comment (arthurvr) [inline README.md]: `3.`

File: README.md [hunk 10]
```diff
@@ -160,15 +205,14 @@ In order to guarantee that the latest version of your Service Worker script is b
...
+1. Navigate to `chrome://serviceworker-internals` and unregister any Service Workers registered by Polymer Starter Kit for your app just in case there's a copy of it cached.
```
Comment (arthurvr) [inline README.md]: `4.`

File: app/index.html [hunk 11]
```diff
@@ -184,7 +186,8 @@ <h2 class="paper-font-display2">Contact</h2>
                          precache-file="precache.json">
       </platinum-sw-cache>
     </platinum-sw-register>
-
+    -->
+
```
Comment (arthurvr) [inline app/index.html]: I personally don't really like this extra line now.

**Comment by arthurvr** [comment]
I still don't like the fact that we're, IMHO, making the experience for the 'raw PSK' users less good just because it will be easier for the yeoman generator.

This also needs a rebase.

File: README.md (hunk 5 above)
Comment (chuckh) [inline README.md]: With markdown you just put #. such 1. for each numbered item so identified it as a numbered item, like - for bulleted item.   So you don't have to go 1., 2., 3., just 1.,1.,1..

**Comment by chuckh** [comment]
> I still don't like the fact that we're, IMHO, making the experience for the 'raw PSK' users less good just because it will be easier for the yeoman generator.

@arthurvr I don't agree that it makes it less good for raw PSK users. Since we are changing PSK by default not to enable Service workers, the code needs to commented and instructions are needed to show how to enable.

I realize this needs to be rebased. I am having issues doing so. Any suggestions on the best approach are appreciated.

File: README.md [hunk 12]
```diff
@@ -118,9 +126,56 @@ Polymer uses [Bower](http://bower.io) for package management. This makes it easy
...
+  <!-- Uncomment next block to enable Service Worker Support (2/2) -->
+  <!--
+  <link rel="import" href="../bower_components/platinum-sw/platinum-sw-cache.html">
+  <link rel="import" href="../bower_components/platinum-sw/platinum-sw-register.html">
+  -->
+  ```
+1. Add 'precache' to the list in the 'default' gulp task like below.
+[(gulpfile.js)](https://github.com/PolymerElements/polymer-starter-kit/blob/master/gulpfile.js)  
+  
+  ```c
```
Comment (arthurvr) [inline README.md]: the leading two spaces is against what we're generally doing. Also needs to change `c` to the correct language.

**Comment by addyosmani** [comment]
@arthurvr Some context: I'm mostly ignoring the need of the Yeoman generator for this particular change - you're right that PSK should absolutely have solid reasons for our direct users for such a change-up. Instead, I'm basing whether we should comment out this feature on the level of developer confusion having Service Worker enabled may cause. 

One comment I've seen a few times is that developers unfamiliar with the debugging experience for SW in Chrome run into an issue where anything else served on the same port (after running/closing PSK) still caches content. This is entirely down to browsers like Chrome needing to up the debugging experience around this web platform feature.

If you're not familiar with how to unregister/clear out your SW caches, there's a bit of a ramp up that might cause those new to Polymer from getting the most out of it. That said, commenting out SW isn't a foregone conclusion. I just want to make sure if we do comment it out, it's for reasons related to making the experience easier for everyone. Making it feasible for third-party tools to consume us and offer SW as an option is if anything a side-effect imo.

**Comment by chuckh** [comment]
@addyosmani I have try to rebase and squash these commits without any luck. I keep coming up against conflict with the readme.md commit where you added Neon Elements info.

Changes with this pull request includes your Neon Elements info, just not all the commits squashed. 

So can you try rebasing then merging or just merging this. I hope I don't have to start over from scratch with this pull request.  

Thanks, Chuck

**Comment by addyosmani** [comment]
@chuckh I've tried pulling your commits into a fresh PR and squashing but am running into the same conflict you did. Can fix up, but may lose some original commit authorship. Would you prefer to create a fresh PR against master or go ahead with the above?

**Comment by chuckh** [comment]
@addyosmani I will go ahead and create a fresh pull request against master.

**Comment by addyosmani** [comment]
:+1:

**Comment by chuckh** [comment]
Pull request #191 replaces this pull request.

**Pull request closed without merging**